### PR TITLE
feat: GA4GH passport demonstration

### PR DESCRIPTION
**Pull request by andrewpatto**
Not for merging.

Just a quick demonstration that we can deploy of passports as Auth headers.

### CODE REVIEW

File: htsget-http/src/http_core.rs [hunk 1]
```diff
@@ -23,11 +26,234 @@ pub async fn get(
...
+    // OIDC discovery and JWKS
+    // e.g. load and cache https://broker-australia.aai.dev.umccr.org/jwks
+    let e = "AQAB";
+
+    // we store a map of issuer -> public RSA "n"
+    let brokers = HashMap::from([
+        ("https://broker-europe.aai.dev.umccr.org", DecodingKey::from_rsa_components("n5POaIPBp04XYkujy7ILkeYpuqPtzRz6fWNFZy7fR6qPLycP4aANFA2xRjr5YP1XXRnm7Jg23gmSbGYFNlnKDRNf67PM53L9Afx56DAUufH0vAISOq2e-i2P4aWZCGcc-d-8tmNTQ3FFcS2wD3bwUsVG2uLXVcdHvmvbTVVIXYxNiznXLk3sNBjuL40VIKEK_x8KSX04_0_x07KKFW1rqj1sguzBeF-NJRTGKuplFEwVM5TxAXRNQe1VeC3_TAEK4PRD8bzzFBz3y-fyovlppfjeOEbIlLT4mafzD130dlINw4xdaLQIPkQb8UE8O-XNKUzguSdUOw0TYB49mFIm5w", e).unwrap()),
+        ("https://broker-australia.aai.dev.umccr.org", DecodingKey::from_rsa_components("q-mBB5jBJCCc68ALWF_A-zOM4S5gsKPB6qeFYWe_uzkfWf-jcSCHRrRCsMRzflVvZbz3mbBmqp8FVOnEWQe62x0qdNSZRUmRWkeBKhi0yxjXbKV7e11Sv5XWxxGhYL-gYzJXqQLR9T8ZfcDeQvEtobznm51VkZ1UgD6QogjtpCK-LL3t5NK2wS5lZO3K5GM4spbnXOLbbUU0sRHKujkYa6frY71i3EAs_nrzkTRmT0I_QkE24XlGRh0zbM67pW1n9SKHsEpFIEzXy3ebBfHhaWKdxY4qhGqbOvft-rgNGGAXPEkKbygIcE0Uif1DvaNHD6KDAeP0DGJVr3qCKQ4Naw", e).unwrap()),
+      ]);
```
Comment (andrewpatto) [inline htsget-http/src/http_core.rs:92]: yes those are keys! they are public keys for known RSA keys though..

**Comment by andrewpatto** [comment]
asking for review *not* for merging purposes.. just let me know anything I've done wildly wrong.

If ok - can we have this branch deployed to UoM demo?

**Comment by brainstorm** [comment]
Quite hardcore/hardcoded changes not to be merged to main, but I see your intent. Changes look fine within those premises, so I'd say go ahead and deploy to UoM demo 🚀

**Comment by andrewpatto** [comment]
Yes to get rid of the embedded keys we'd need to instigate a new config mechanism - and probably move the either auth into some sort of plugin architecture. Which is a bit more than I want for this demo.

I don't know how to deploy to demo btw. (I mean, is it literally the CDK in deploy?)

**Review by mmalenic — APPROVED** [review]
Yes, looks good. For deploying, it should just be a CDK deploy in the `deploy` directory to the demo account. Although the public property needs to be enabled in the settings.ts

**Review by brainstorm — APPROVED** [review]

**Comment by brainstorm** [comment]
Thanks for this Andrew! Closing this for now as we'll not be merging and keeping the branch around so that we can revisit this and implement/merge once GA4GH impls (and elsewhere upstream) are a bit more stable? Probably worth looking at it again once you're back from Sweden?

**Comment by andrewpatto** [comment]
Yes no problems - I am going to grab the code and possibly put it in a standalone GA4GH repo of "passport decoders" (in different languages). So it won't need to have any htsget tie in..
But yes - should re-visit later specifically for htsget

**Pull request closed without merging**